src/PrestaShopBundle/Form/FormBuilderModifier.php

```diff
@@ -0,0 +1,116 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form;
+
+use Symfony\Component\Form\Exception\InvalidArgumentException;
+use Symfony\Component\Form\FormBuilderInterface;
+
+class FormBuilderModifier
+{
+    /**
+     * @var FormBuilderInterface
+     */
+    private $formBuilder;
+
+    /**
+     * @param FormBuilderInterface $formBuilder
+     * @param string $targetFieldName
+     * @param string|FormBuilderInterface $newChild
+     * @param string|null $type
+     * @param array $options
+     */
+    public function addAfter(FormBuilderInterface $formBuilder, string $targetFieldName, $newChild, ?string $type = null, array $options = []): void
+    {
+        $this->formBuilder = $formBuilder;
+        $this->assertFieldExists($targetFieldName);
+        $formChildren = $this->cleanAllChildren();
+
+        foreach ($formChildren as $childName => $formType) {
+            $this->formBuilder->add($formType);
+            if ($childName === $targetFieldName) {
+                $this->formBuilder->add($newChild, $type, $options);
+            }
+        }
+    }
+
+    /**
+     * @param FormBuilderInterface $formBuilder
+     * @param string $targetFieldName
+     * @param string|FormBuilderInterface $newChild
+     * @param string|null $type
+     * @param array $options
+     */
+    public function addBefore(FormBuilderInterface $formBuilder, string $targetFieldName, $newChild, ?string $type = null, array $options = []): void
+    {
+        $this->formBuilder = $formBuilder;
+        $this->assertFieldExists($targetFieldName);
+        $formChildren = $this->cleanAllChildren();
+
+        foreach ($formChildren as $childName => $formType) {
+            if ($childName === $targetFieldName) {
+                $this->formBuilder->add($newChild, $type, $options);
+            }
+            $this->formBuilder->add($formType);
+        }
+    }
+
+    /**
+     * @return array
+     */
+    private function cleanAllChildren(): array
+    {
+        $formTypes = [];
+        foreach ($this->formBuilder->all() as $formType) {
+            $typeName = $formType->getName();
+            // collect all the form child into local variable and remove them from
+            $formTypes[$typeName] = $formType;
+            $this->formBuilder->remove($typeName);
+        }
+
+        return $formTypes;
+    }
+
+    /**
+     * @param string $name
+     */
+    private function assertFieldExists(string $name): void
+    {
+        if ($this->formBuilder->has($name)) {
+            return;
+        }
+
+        throw new InvalidArgumentException(
+            sprintf(
+                'Form field "%s" does not exist in "%s" form',
+                $name,
+                $this->formBuilder->getName()
+            )
+        );
+    }
```

Comment:
I wonder if `formBuilder` should be a parameter 🤔

Reply:
Actually, I used this to avoid duplicating the same parameter everywhere, but I understand your point I'm gonna modify this 😉

Reply:
In this case you should find a way to initialize this variable in the constructor, because the developer who wants to use this method will have to always initialize this attribute before, and as it is not indicated anywhere, the risk of forgetting is very high which will trigger errors like:

`Fatal error: Call to a member function has() on null` or `Fatal error: Call to a member function getName() on null`

Reply:
No, because I don't want it to be coupled with the form builder, or it means you need to create a new one each time you want to make a modification With this "parameter way" it can be defined as a service and be reused

And there was no risk of having null because the function set the inner field each times it's called, but there are more risks to have stateful bugs, so I removed it anyway

Reply:
Look good for me now ! Thanks